sdl2promela/stop_condition/translator.py

```diff
@@ -278,36 +291,165 @@ def _generate(context: GenerateContext, expr: model.VariableReference):
     It means that variable is a selector of CHOICE.
     """
     if context.choice_selection is not None:
+        # The list of available alternatives of CHOICE is case-sensitive
+        # Stop Condition Language is case-insensitive,
+        # so, the existing identifier should be generated.
+        selected_alternatives = [
+            alternative
+            for alternative in context.choice_selection_alternatives
+            if alternative.lower() == expr.name.lower()
+        ]
+        if len(selected_alternatives) != 1:
+            raise TranslateException(
+                f"The CHOICE datatype '{context.choice_selection}' has no alternative with name '{expr.name}'"
+            )
+
         return promelaBuilder.VariableReferenceBuilder(
-            context.choice_selection + "_" + expr.name + "_PRESENT"
+            context.choice_selection + "_" + selected_alternatives[0] + "_PRESENT"
         ).build()
     else:
         return promelaBuilder.VariableReferenceBuilder(expr.name).build()
 
 
+class FirstVariableInfo:
```

Comment:
I think that the concept of a first variable should be better explained... and what it means that variable consumes elements...

Reply:
I've added more comments, but in the function _get_first_variable